drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -398,10 +400,17 @@ static int esp32_wifi_scan(const struct device *dev, scan_result_cb_t cb)
 		return -EINPROGRESS;
 	}
 
+	if (!params) {
+		return -EINVAL;
```

Comment:
This breaks compatibility with current applications, as this is an optional param, IMHO we should just ignore.